runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6372,7 +6372,202 @@ TR_J9ByteCodeIlGenerator::loadFromCP(TR::DataType type, int32_t cpIndex)
             loadSymbol(TR::dload, symRefTab()->findOrCreateDoubleSymbol(_methodSymbol, cpIndex));
          break;
       case TR::Address:
-         if (method()->isClassConstant(cpIndex))
+         if (method()->isConstantDynamic(cpIndex))
+            {
+            bool isCondyUnresolved = _methodSymbol->getResolvedMethod()->isUnresolvedConstantDynamic(cpIndex);
+            J9UTF8 *returnTypeUtf8 = (J9UTF8 *)_methodSymbol->getResolvedMethod()->getConstantDynamicTypeFromCP(cpIndex);
+            int returnTypeUtf8Length = J9UTF8_LENGTH(returnTypeUtf8);
+            char* returnTypeUtf8Data = (char *)J9UTF8_DATA(returnTypeUtf8);
+            bool isCondyPrimitive = (1 == returnTypeUtf8Length);
+
+            char* symbolTypeSig = NULL;
+            int32_t symbolTypeSigLength = 0;
+            // For non-primitive condy, resolved or not, we create a static symbol,
+            // store the class info in the symbol so it may be retrieved by optimizer later via
+            // getTypeSignature.
+            if (!isCondyPrimitive)
+               {
+               symbolTypeSig = (char*)comp()->trMemory()->allocateMemory(returnTypeUtf8Length, heapAlloc);
+               symbolTypeSigLength = returnTypeUtf8Length;
+               memcpy(symbolTypeSig, returnTypeUtf8Data, symbolTypeSigLength);
+               }
+
+            TR_OpaqueClassBlock * typeClassBlock = NULL;
+            int32_t valueOffset = 0;
+
+            // If condy is primitive type and resolved, load the primitive constant;
+            // Otherwise, load using a CP symol (for resolved and unresolved object type),
+            // and generate subsequent loadi for the unresolved primitive 'value' field if needed (because
+            // for unresolved primitive the resovle helper only returns an autobox'd object).
+            if (isCondyPrimitive)
+               {
+               char *autoboxClassSig = NULL;
+               int32_t autoboxClassSigLength = 0;
+               switch (returnTypeUtf8Data[0])
```

Comment:
There should be an assert above this to make sure the signature is valid - ie 1 char.

Reply:
Here we are inside the `if (isCondyPrimitive)` condition so it's already implicitly ensured that `turnTypeUtf8Length` is 1 for the type signature UTF8 : `bool isCondyPrimitive = (1 == returnTypeUtf8Length);`